src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -271,29 +278,76 @@ export class IntegrationListDirectory extends AsyncComponent<
     );
   }, TEXT_SEARCH_ANALYTICS_DEBOUNCE_IN_MS);
 
+  /**
+   * Get filter parameters and guard against `queryString.parse` returning arrays.
+   */
+  getFilterParameters = (): {searchInput: string; selectedCategory: string} => {
+    const {category, search} = queryString.parse(this.props.location.search);
+
+    const selectedCategory = Array.isArray(category) ? category[0] : category || '';
+    const searchInput = Array.isArray(search) ? search[0] : search || '';
+
+    return {searchInput, selectedCategory};
+  };
+
+  /**
+   * Update the query string with the current filter parameter values.
+   */
+  updateQueryString = () => {
+    const {searchInput, selectedCategory} = this.state;
+
+    const searchString = queryString.stringify({
+      ...queryString.parse(this.props.location.search),
+      search: searchInput ? searchInput : undefined,
+      category: selectedCategory ? selectedCategory : undefined,
+    });
+
+    browserHistory.replace({
+      pathname: this.props.location.pathname,
+      search: searchString ? `?${searchString}` : undefined,
+    });
+  };
+
+  /**
+   * Filter the integrations list by ANDing together the search query and the category select.
+   */
+  calculateDisplayedList = (): AppOrProviderOrPlugin[] => {
```

Comment:
@mgaeta I think `calculateAndSetDisplayedList` would be a better name

Reply:
I think I'll stick with "update" to match the other functions in this file.